python/iterating_for_each_two_elements.md

```diff
@@ -0,0 +1,28 @@
+# Interating a list for each 2 elements using `zip` and `iter`
+
+The `zip` function is used to return a list of tuples from each of the iterables passed to it
+
+```python
+  >>> list_one = ['a', 'b']
+  >>> list_two = ['c', 'd']
+  >>> zip(list_one, list_two)
```

Comment:
Is it also worth us showing what happens when you zip a list with _itself_ without using `iter` to show the different behaviour?

```python
>>> list = [1, 2, 3]
>>> zip(list, list)
[(1, 1), (2, 2), (3, 3)]
```